src/api/java/com/minecolonies/api/colony/requestsystem/StandardRequestManager.java

```diff
@@ -0,0 +1,1274 @@
+         *
+         * @param provider The removed provider.
+         * @throws IllegalArgumentException is thrown when no provider with the same token is registered.
+         */
+        @Override
+        public void onProviderRemovedFromColony(@NotNull final IRequestResolverProvider provider) throws IllegalArgumentException
+        {
+            wrappedManager.onProviderRemovedFromColony(provider);
+        }
+
+        @Override
+        public NBTTagCompound serializeNBT()
+        {
+            return wrappedManager.serializeNBT();
+        }
+
+        @Override
+        public void deserializeNBT(final NBTTagCompound nbt)
+        {
+            wrappedManager.deserializeNBT(nbt);
+        }
+    }
+
+    /**
+     * Class used to handle internal reassignment changes.
+     * Take the given blacklist into account when it assigns the requests.
+     */
+    private final static class WrappedBlacklistAssignmentRequestManager extends AbstractWrappedRequestManager
+    {
+
+        @NotNull
+        private final Collection<IToken> blackListedResolvers;
+
+        private WrappedBlacklistAssignmentRequestManager(@NotNull final StandardRequestManager wrappedManager, @NotNull final Collection<IToken> blackListedResolvers)
+        {
+            super(wrappedManager);
+            this.blackListedResolvers = blackListedResolvers;
+        }
+
+        /**
+         * Method used to assign a request to a resolver.
+         *
+         * @param token The token of the request to assign.
+         * @throws IllegalArgumentException when the token is not registered to a request, or is already assigned to a resolver.
+         */
+        @NotNull
+        @Override
+        public void assignRequest(@NotNull final IToken token) throws IllegalArgumentException
+        {
+            RequestHandler.assignRequest(wrappedManager, RequestHandler.getRequest(wrappedManager, token), blackListedResolvers);
+        }
+    }
+
+    /**
+     * Class used to handle internal state changes that might cause a loop.
+     * Simply returns without notifying its wrapped manager about the state change.
+     */
+    private final static class WrappedStaticStateRequestManager extends AbstractWrappedRequestManager
+    {
+
+        public WrappedStaticStateRequestManager(@NotNull final StandardRequestManager wrappedManager)
+        {
+            super(wrappedManager);
+        }
+
+        @NotNull
+        @Override
+        /**
+         * Method to update the state of a given request.
+         *
+         * @param token The token that represents a given request to update.
+         * @param state The new state of that request.
+         * @throws IllegalArgumentException when the token is unknown to this manager.
+         */
+        public void updateRequestState(@NotNull final IToken token, @NotNull final RequestState state) throws IllegalArgumentException
+        {
```

Comment:
To implement in the future. @Asherslab add: //TODO: Implement when link is created with workers.